scripts/sidekick/sidekick.js

```diff
@@ -0,0 +1,62 @@
+async function getContentSourceUrl(owner, repo, ref) {
+  const res = await fetch(`https://admin.hlx.page/sidekick/${owner}/${repo}/${ref}/env.json`);
+  if (!res || !res.ok) {
+    return null;
+  }
+  const env = await res.json();
+  if (!env) {
+    return null;
+  }
+  return env.contentSourceUrl;
+}
+async function openAemEditor(event) {
+  const { owner, repo, ref } = event.detail.data.config;
+  const contentSourceUrl = await getContentSourceUrl(owner, repo, ref);
+  const path = window.location.pathname;
+  const editorUrl = `${contentSourceUrl}${path}?cmd=open`;
+  // open the editor in a new tab
+  window.open(editorUrl, '_blank');
+}
+
+function getButton(sk, selector) {
+  let btn = sk.shadowRoot.querySelector(selector);
+  if (btn) {
+    return btn;
+  }
+  return new Promise((resolve) => {
+    const check = () => {
+      btn = sk.shadowRoot.querySelector(selector);
+      if (btn) {
+        resolve(btn);
+      } else {
+        setTimeout(check, 100);
+      }
+    };
+    check();
+  });
+}
+
+async function overrideEditButton(sk) {
+  const oldEditBtn = await getButton(sk, '.edit.plugin');
+  const newEditBtn = await getButton(sk, '.aemedit.plugin');
+  oldEditBtn.replaceWith(newEditBtn);
```

Comment:
why do we need a new button actually? can we get the old button and just replace the click behaviour?

Reply:
I tried it but it didn't work because:
- the Edit button is disabled
- even if we enable it, the ootb logic is triggered when the button is clicked and it cannot be overridden